x/auth/middleware/basic.go

```diff
@@ -0,0 +1,330 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/codec/legacy"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/multisig"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+type validateBasicTxHandler struct {
+	next tx.Handler
+}
+
+// ValidateBasicMiddleware will call tx.ValidateBasic, msg.ValidateBasic(for each msg inside tx)
+// and return any non-nil error.
+// If ValidateBasic passes, middleware calls next middleware in chain. Note,
+// validateBasicTxHandler will not get executed on ReCheckTx since it
+// is not dependent on application state.
+func ValidateBasicMiddleware(txh tx.Handler) tx.Handler {
+	return validateBasicTxHandler{
+		next: txh,
+	}
+}
+
+var _ tx.Handler = validateBasicTxHandler{}
+
+// CheckTx implements tx.Handler.CheckTx.
+func (basic validateBasicTxHandler) CheckTx(ctx context.Context, tx sdk.Tx, req abci.RequestCheckTx) (abci.ResponseCheckTx, error) {
```

Comment:
nit: I would calling the receiver `basic`. Instead, consider something more concise such as `h` or `txh`.